frontend/src/components/Map/index.js

```diff
@@ -51,6 +52,19 @@ export default function Map({ draggable = true }) {
       });
   }, [location, map]);
 
+  const addLegend = (data) => {
+    const clusters = data.clusters;
+    const colorGroups = data.groups;
+    const newRanges =
+      clusters &&
+      clusters.map((range, i) => {
+        return {
+          label: `${range[0].toLocaleString()} - ${range[1].toLocaleString()}`,
```

Comment:
See, here's were we could be doing the native rounding I was speaking about before with either [this](https://manas.slack.com/archives/C013E1LMBQE/p1592402876015800?thread_ts=1592350773.007400&cid=C013E1LMBQE) native JS thing I mentioned before, or actually, [passing some options like `maximumSignificantDigits`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString) to the `toLocaleString()` we're already using!